massa-execution/src/types.rs

```diff
@@ -26,26 +27,53 @@ pub(crate) struct StepHistoryItem {
 }
 
 #[derive(Clone)]
-/// Stateful context, providing an execution context to massa between module executions
+/// Stateful context, providing a context during the execution of a module
 pub(crate) struct ExecutionContext {
+    /// final and active ledger at the current step
     pub ledger_step: SCELedgerStep,
+
+    /// max gas for this execution
     pub max_gas: u64,
+
+    /// coins transferred to the target address during a call, stacked
     pub coins_stack: Vec<Amount>,
+
+    /// gas price of the execution
     pub gas_price: Amount,
+
+    /// slot at which the execution happens
     pub slot: Slot,
+
+    /// counter of newly created addresses so far during this execution
     pub created_addr_index: u64,
+
+    /// block ID, if one is present at this slot
     pub opt_block_id: Option<BlockId>,
+
+    /// block creator addr, if there is a block at this slot
     pub opt_block_creator_addr: Option<Address>,
+
+    /// address call stack, most recent is at the back
     pub call_stack: VecDeque<Address>,
+
+    /// list of addresses created so far during excution, stacked
```

Comment:
Why is it a vec of vecs ?

Reply:
because it's a vec<address> that is stacked

Reply:
I would add something like "stacked,in the same order as the call stack"

By the way, id module 1 calls module 2 calls module 3, and addresses are created only during module2 will this be the same length as the call stack ?

Reply:
> That's probably an issue for another day, but how do you ensure that coin_stack and owned address stack are corresponding to the call stack ?
> 
> Maybe the call stack could be somthing like a VecDeque<(Address, coins: Amount, owned_addrs: Vec
> ) Or am I missing something ?

Very good idea. Will refactor it as  `call_stack: Vec<StackElement>`:

```rust
struct StackElement {
    address: Address,
    coins: Amount,
    owned_addrs: Set<Address>
}
```